src/pke/include/scheme/ckksrns/ckksrns-schemeswitching.h

```diff
@@ -0,0 +1,420 @@
+                           NativeVector* nativeVec) const;
+
+#if NATIVEINT == 128 && !defined(__EMSCRIPTEN__)
+    /**
+   * Set modulus and recalculates the vector values to fit the modulus
+   *
+   * @param &vec input vector
+   * @param &bigValue big bound of the vector values.
+   * @param &modulus modulus to be set for vector.
+   */
+    void FitToNativeVector(uint32_t ringDim, const std::vector<__int128>& vec, __int128 bigBound,
+                           NativeVector* nativeVec) const;
+
+    constexpr __int128 Max128BitValue() const {
+        // 2^127-2^73-1 - max value that could be rounded to int128_t
+        return ((unsigned __int128)1 << 127) - ((unsigned __int128)1 << 73) - (unsigned __int128)1;
+    }
+
+    inline bool is128BitOverflow(double d) const {
+        const double EPSILON = 0.000001;
+
+        return EPSILON < (std::abs(d) - Max128BitValue());
+    }
+#else  // NATIVEINT == 64
+    constexpr int64_t Max64BitValue() const {
+        // 2^63-2^9-1 - max value that could be rounded to int64_t
+        return 9223372036854775295;
+    }
+
+    inline bool is64BitOverflow(double d) const {
+        const double EPSILON = 0.000001;
+
+        return EPSILON < (std::abs(d) - Max64BitValue());
+    }
+#endif
+
+    //------------------------------------------------------------------------------
+    // Private members
+    //------------------------------------------------------------------------------
+
+    // the LWE cryptocontext to generate when scheme switching from CKKS
+    BinFHEContext m_ccLWE;
+    // the CKKS cryptocontext for the intermediate modulus switching in CKKS to FHEW
+    CryptoContext<DCRTPoly> m_ccKS;
+    // the associated ciphertext modulus Q for the LWE cryptocontext
+    NativeInteger m_modulus_LWE;
+    // the target ciphertext modulus Q for the CKKS cryptocontext. We assume the switching goes to the same initial cryptocontext
+    NativeInteger m_modulus_CKKS_initial;
+    // the ciphertext modulus Q' for the CKKS cryptocontext that is secure for the LWE ring dimension
+    NativeInteger m_modulus_CKKS_from;
+    // switching key from CKKS to FHEW
+    EvalKey<DCRTPoly> m_CKKStoFHEWswk;
+    // switching key from FHEW to CKKS
+    Ciphertext<DCRTPoly> m_FHEWtoCKKSswk;
+    // a ciphertext under the intermediate cryptocontext
+    Ciphertext<DCRTPoly> m_ctxtKS;
+    // number of slots encoded in the CKKS ciphertext
+    uint32_t m_numSlotsCKKS;
+    // baby-step dimensions for linear transform for CKKS->FHEW, FHEW->CKKS
+    uint32_t m_dim1CF;
+    uint32_t m_dim1FC;
+    // starting levels for linear transforms
+    uint32_t m_LCF;
+    uint32_t m_LFC;
+
+    // target FHEW plaintext modulus
+    uint64_t m_plaintextFHEW;
+    // scaling factor of CKKS "outer" ciphertext
+    double m_scFactorOuter;
+
+#define Pi 3.14159265358979323846
+
+    // K = 16
+    const std::vector<double> g_coefficientsFHEW16{
```

Comment:
would it make sense to use constexpr std::array<> for these?

Reply:
I did, please check. EvalChebyShevSeries takes as argument a std::vector<>, so I populate it with the array as follows
coefficientsFHEW.insert(coefficientsFHEW.end(), &g_coefficientsFHEW128_8[0], &g_coefficientsFHEW128_8[LEN_128_8])